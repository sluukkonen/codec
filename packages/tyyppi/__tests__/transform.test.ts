import { fromJson, string, transform } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

const trimmedString = transform(
  string,
  (s) => s.trim(),
  (s) => s,
)

test("should run the parsed value through the transformation function", () => {
  expectParseSuccess(trimmedString, "   abc", "abc")
})

test("should fail if the underlying codec fails", () => {
  expectParseFailure(trimmedString, null)
})

test("should work with non-simple codecs", () => {
  const plus1 = transform(
    fromJson.bigint,
    (n) => n + 1n,
    (n) => n - 1n,
  )
  expectParseSuccess(plus1, "1", 2n)
})
