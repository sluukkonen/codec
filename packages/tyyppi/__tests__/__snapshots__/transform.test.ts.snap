// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should fail if the underlying codec fails 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected a string, received null",
    "path": [],
    "received": "null",
  },
]
`;
